### ISSUE TITLE
bug(mat-form-field): Accessibility issue with input placeholders

### ISSUE THREAD

**Issue by yann-armelin-1a** [angular/components#23858]
#### Reproduction

Create a matInput inside mat-form-field, with any light theme:
https://stackblitz.com/edit/angular-jvcyas

( all appearances are impacted )

#### Expected Behavior

To comply with WCAG AA, contrast between text color and background should always be higher than 4.5:1, even for placeholders.

#### Actual Behavior

Using `outline` or `standard` the contrast is `3.03:1`, and with `outline` it is `3:1` :
![image](https://user-images.githubusercontent.com/61547087/139444106-9ac848de-9714-4adc-8b95-1e91a0b4dcf2.png)

There are several other contrast issues with default themes, but most of them can be addressed using a custom theme.
Unfortunately placeholders colors are hardcoded to `rgba(0, 0, 0, 0.42)`, and not customizable via themes:
https://github.com/angular/components/blob/202c667e3439e7412fed981993aa51ee2b0f41f6/src/material/core/style/_form-common.scss#L20

Using `$dark-secondary-text` instead  , that is hardcoded to rgba(0, 0, 0, 0.54) , would provide a compliant contrast (or be very close to it):
https://github.com/angular/components/blob/202c667e3439e7412fed981993aa51ee2b0f41f6/src/material/core/theming/_palette.scss#L30

On `outline` or `standard` => `4.6:1`
On `fill` => `4.47:1`

#### Environment

- Angular: 12
- CDK/Material: 12
- Browser(s): All
- Operating System (e.g. Windows, macOS, Ubuntu):  All

**Comment by sergiotome** [comment]
Is there any update on this? We are hearing similar concerns with color contrast for MatInput

**Comment by crisbeto** [comment]
We can't really change the defaults since they come from Material, but it should be easy to override now with the styling API.
